internal/driver/driver.go

```diff
@@ -0,0 +1,421 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+//
+// Copyright (C) 2022 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package driver
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+	"sync"
+	"time"
+
+	sdkModel "github.com/edgexfoundry/device-sdk-go/v2/pkg/models"
+	sdk "github.com/edgexfoundry/device-sdk-go/v2/pkg/service"
+	"github.com/edgexfoundry/go-mod-bootstrap/v2/bootstrap/secret"
+	"github.com/edgexfoundry/go-mod-bootstrap/v2/bootstrap/startup"
+	"github.com/edgexfoundry/go-mod-bootstrap/v2/config"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/clients/logger"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/errors"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/models"
+
+	"github.com/IOTechSystems/onvif"
+	"github.com/IOTechSystems/onvif/device"
+	wsdiscovery "github.com/IOTechSystems/onvif/ws-discovery"
+)
+
+var once sync.Once
+var driver *Driver
+
+const (
+	URLRawQuery = "urlRawQuery"
+	jsonObject  = "jsonObject"
+)
+
+// Driver implements the sdkModel.ProtocolDriver interface for
+// the device service
+type Driver struct {
+	lc            logger.LoggingClient
+	asynchCh      chan<- *sdkModel.AsyncValues
+	deviceCh      chan<- []sdkModel.DiscoveredDevice
+	config        *configuration
+	lock          *sync.RWMutex
+	deviceClients map[string]*DeviceClient
+}
+
+// NewProtocolDriver initializes the singleton Driver and
+// returns it to the caller
+func NewProtocolDriver() *Driver {
+	once.Do(func() {
+		driver = new(Driver)
+		driver.deviceClients = make(map[string]*DeviceClient)
+	})
+
+	return driver
+}
+
+// Initialize performs protocol-specific initialization for the device
+// service.
+func (d *Driver) Initialize(lc logger.LoggingClient, asyncCh chan<- *sdkModel.AsyncValues,
+	deviceCh chan<- []sdkModel.DiscoveredDevice) error {
+	d.lc = lc
+	d.asynchCh = asyncCh
+	d.deviceCh = deviceCh
+	d.lock = new(sync.RWMutex)
+
+	camConfig, err := loadCameraConfig(sdk.DriverConfigs())
+	if err != nil {
+		panic(fmt.Errorf("load camera configuration failed: %w", err))
+	}
+	d.config = camConfig
+
+	deviceService := sdk.RunningService()
+
+	for _, device := range deviceService.Devices() {
+		d.lc.Infof("Initializing device client for '%s' camera", device.Name)
+
+		deviceClient, err := NewDeviceClient(device, d.config, d.lc)
```

Comment:
```suggestion
		onvifClient, err := NewOnvifClient(device, d.config, d.lc)
```
Apply similar changes through the code

Reply:
Modified.